File: client/src/command/vanilla/util/declarations/DblCommand.ts
import { ApplicationCommandOptionTypes } from "discordeno/types";
import { createCommand } from "../../../structures/createCommand";
import DblExecutor from "../DblExecutor";

const DblCommand = createCommand({
    name: 'dbl',
    description: '[Utils] Vote Foxy on top.gg',
    descriptionLocalizations: {
        "pt-BR": '[Utils] Vote na Foxy no top.gg'
    },
    category: 'util',
    options: [
        {
            name: "upvote",
            description: "[Utils] Vote for Foxy on top.gg",
            descriptionLocalizations: {
                "pt-BR": "[Utilitários] Vote na Foxy no top.gg"
            },
            type: ApplicationCommandOptionTypes.SubCommand
        }
    ],

    execute: async (context, endCommand, t) => {
        DblExecutor(context, endCommand, t);
    }
});

export default DblCommand;

File: foxy/src/main/kotlin/net/cakeyfox/foxy/command/vanilla/utils/declarations/DblCommand.kt
package net.cakeyfox.foxy.command.vanilla.utils.declarations

import net.cakeyfox.foxy.command.structure.FoxyCommandDeclarationWrapper
import net.cakeyfox.foxy.command.vanilla.utils.DblExecutor
import net.dv8tion.jda.api.interactions.IntegrationType
import net.dv8tion.jda.api.interactions.InteractionContextType

class DblCommand : FoxyCommandDeclarationWrapper {
    override fun create() = command(
        "dbl",
        "dbl.description",
        category = "utils",
        interactionContexts = listOf(
            InteractionContextType.BOT_DM,
            InteractionContextType.GUILD,
            InteractionContextType.PRIVATE_CHANNEL
        ),

        integrationType = listOf(
            IntegrationType.GUILD_INSTALL,
            IntegrationType.USER_INSTALL
        )
    ) {
        subCommand(
            "vote",
            "dbl.vote.description",
            baseName = this@command.name,

            block = {
                executor = DblExecutor()
            }
        )
    }
}